c-bindings-gen/src/types.rs

```diff
@@ -225,6 +226,13 @@ pub enum DeclType<'a> {
 	EnumIgnored,
 }
 
+// templates_defined is walked to write the C++ header, so if we use the default hashing it get
+// reordered on each genbindings run. Instead, we use SipHasher (which defaults to 0-keys) so that
+// the sorting is stable across runs. It is deprecated, but the "replacement" doesn't actually
+// accomplish the same goals, so we just ignore it.
+#[allow(deprecated)]
+type NonRandomHash = hash::BuildHasherDefault<hash::SipHasher>;
```

Comment:
Could this same thing be accomplished by collecting the `HashMap`'s keys in a `Vec`, sorting them, then iterating over the vector and looking up entries in the map? Then the order would both stable and somewhat sensible to someone reading the header.

Reply:
I'm not sure that alphabetical is much more sensible than random - usage order would be nice, but for that we'd have to not use a HashMap and I didn't look into if that's practical. At the end of the day the set of things in the file is huge, and they're all identical, so I'm not sure anyone is ever gonna read lightningpp.hpp instead of just inferring it from lightning.h.